src/test/java/com/commercetools/sync/products/utils/ProductUpdateActionUtilsTest.java

```diff
@@ -359,4 +363,58 @@ void buildAddVariantUpdateActionFromDraft_WithMultipleAssets_BuildsMultipleAddAs
 
         assertThat(result).containsExactlyElementsOf(expectedActions);
     }
+
+    @Test
+    void getAllVariants_WithNoVariants_ShouldReturnListWithNullMasterVariant() {
+        final ProductDraft productDraft = ProductDraftBuilder
+            .of(mock(ProductType.class), ofEnglish("name"), ofEnglish("slug"), emptyList())
+            .build();
+
+        final List<ProductVariantDraft> allVariants = getAllVariants(productDraft);
+
+        assertThat(allVariants).hasSize(1).containsOnlyNulls();
```

Comment:
Can you have null? The code will not work if you have null, right?

Reply:
resolved f2f2.